#257 get rid of start page; assert error messages of browser.conditions

aleksandr-kotlyar opened the pull request:
Resolves #257

aleksandr-kotlyar committed:
#257 get rid of start page; add assertions for error messages of browser.conditions

aleksandr-kotlyar committed:
#257 raise up codecov of `Browser().open()` method

aleksandr-kotlyar commented:
Code coverage has fallen down because of method `open()` of custom `Browser()`.
It will be fixed in acceptance tests.
![image](https://user-images.githubusercontent.com/17043964/103924485-24ecb880-5127-11eb-896d-8b1821cccc26.png)

aleksandr-kotlyar commented:
> Code coverage has fallen down because of method `open()` of custom `Browser()`.
> It will be fixed in acceptance tests.
> ![image](https://user-images.githubusercontent.com/17043964/103924485-24ecb880-5127-11eb-896d-8b1821cccc26.png)

Raised code coverage back up and little more in two acceptance tests in this PR: https://github.com/yashaka/selene/blob/c36399a67b854eca2c366343612343e34d31616b/tests/acceptance/custom_browser/test_ecosia.py
and
https://github.com/yashaka/selene/blob/c36399a67b854eca2c366343612343e34d31616b/tests/acceptance/shared_browser/test_ecosia.py

yashaka commented on tests/acceptance/custom_browser/conftest.py:
```diff
@@ -0,0 +1,17 @@
+
+
+@pytest.fixture(scope='function')
+def session_browser(session_driver):
```
since this is acceptance kind of test... we should more close to the user style in real life...
in real case, most probably it will be too much to name a browser as session_browser
too long name

let's name it here as just `browser`

aleksandr-kotlyar replied:
okay :) but it will be intersect with imports and juniors could fail a bit if they will try to use both in one module :)

yashaka replied:
should not be a problem
you either pass browser as a fixture to a test function
and don't need an import
and don't need another "browser from selene imports"
or don't pass and use imported browser:)

so kind of it should be pretyy obvious from the context what to use... 

usually it's also one style per project that somebody is going to use...

aleksandr-kotlyar replied:
Agreed. Done it in 33992b29b0efaf88cba3bf37509d24e6b6401ad4

yashaka commented on tests/integration/browser__switch_to__alert_test.py:
```diff
@@ -19,16 +19,16 @@
-    session_browser.open(start_page)
+    GivenPage(session_browser.driver).opened_with_body("""
+        <p>
+        <input id="alert_btn" type="button" onclick="alert('Добрый день')" value="Запустить">
```
I guess here we are not forced to test unicode... So let's stick to English texts...

aleksandr-kotlyar replied:
Agreed. Done it in 26528b1652d2f9192747d0cc72622707cdfb9d4f

yashaka commented on tests/integration/condition__browser__have_url_test.py:
```diff
@@ -19,44 +19,44 @@
+    with pytest.raises(TimeoutException) as err:
         browser.should(have.url('xttp:/'))
-        # TODO: check message too
+    assert "Timed out after 0.1s, while waiting for:" in err.value.msg
```
this is kind of a standart part of error message, that does not depend on actual condition like have.url
so I would miss it... 
this message also helps to explain the context – that it is a "timeout error", but the latter is already reflected in with pytest.raises(TimeoutException)

aleksandr-kotlyar replied:
What about timeout value? Does it count here? Should we leave this message for it?

yashaka replied:
timeout is something that we don't test in this test
and timeout is also a part of "base generic title" of all selene waiting assertions, that should be test in some different kind of test, where we focus exactly on testing "waiting logic"

aleksandr-kotlyar replied:
Agreed. Removed assert timeout part of msg in b2221708789e3b7a1631da963b03a107e00b382b

yashaka commented on tests/integration/condition__browser__have_url_test.py:
```diff
@@ -19,44 +19,44 @@
+    GivenPage(browser.driver).opened_empty()
 
-    with pytest.raises(TimeoutException):
+    with pytest.raises(TimeoutException) as err:
```
is an `err` a standard common way to shorten error in python? 
if yes, then ok
but if no, let's keep it as `error` - it's yet not so long to be so economic :)

aleksandr-kotlyar replied:
i dunno but agree that error is better :) will do.

aleksandr-kotlyar replied:
Agreed. Done it in 35d0033619a81771c5f5b1bc440fe0368d198c51

yashaka commented on tests/integration/condition__browser__have_url_test.py:
```diff
@@ -19,44 +19,44 @@
         browser.should(have.url('xttp:/'))
-        # TODO: check message too
+    assert "Timed out after 0.1s, while waiting for:" in err.value.msg
+    assert "browser.has url" in err.value.msg
```
here it's pretty important to reflect expected value in this assert, not just "has url" but "has url ..."

yashaka replied:
especially taking into account that this part  - rendeing of expected value - may change with the time
when for example we decide to put it in some quotes, etc.

aleksandr-kotlyar replied:
How about move expected url to variable and code will look like:
```
expected_url = 'xttp:/'
browser.should(have.url(expected_url))
assert f"browser.has url {expected_url}" in err.value.msg
```

aleksandr-kotlyar replied:
Done in c7b62b670638b6647d11c9c1c26eef837682efc1

aleksandr-kotlyar committed:
Merge branch 'master' of https://github.com/yashaka/selene into gh_#257_rid_of_start_page

aleksandr-kotlyar committed:
#257 rename 'session_browser' fixture to 'browser'
in case of recommendation https://github.com/yashaka/selene/pull/259/files/c36399a67b854eca2c366343612343e34d31616b#r553893199

aleksandr-kotlyar committed:
#257 changed unicode alert text to english
in case of recommendation https://github.com/yashaka/selene/pull/259/files/c36399a67b854eca2c366343612343e34d31616b#r553894387

aleksandr-kotlyar committed:
#257 rename 'err' to 'error'
as a result of https://github.com/yashaka/selene/pull/259/files/c36399a67b854eca2c366343612343e34d31616b#r553896126

aleksandr-kotlyar committed:
#257 remove assert timeout in error message
as a result of https://github.com/yashaka/selene/pull/259/files/c36399a67b854eca2c366343612343e34d31616b#r553903330

aleksandr-kotlyar committed:
#257 test that expected url value is in error message
as result of review https://github.com/yashaka/selene/pull/259/files/c36399a67b854eca2c366343612343e34d31616b#r553896389

aleksandr-kotlyar committed:
gh #257 remove unnecessary part of exception message
